cmd/server/app/server.go

```diff
@@ -142,6 +143,16 @@ func (p *Proxy) Run(o *options.ProxyRunOptions, stopCh <-chan struct{}) error {
 		defer frontendStop()
 	}
 
+	if o.LeaseLabelSelector != "" || o.LeaseDuration != 0 || o.LeaseGCInterval != 0 || o.LeaseRenewalInterval != 0 {
```

Comment:
This seems redundant after the validation check. Not wrong but we can probably just use o.LeaseLabelSelector != "". Better yet you can add a helper function isLeaseCountingEnabled() which implements this and makes the code more readable. (Especially given that your missing o.LeaseNamespace != "" which suggests the helper would help keep the checks consistent)

Reply:
Added helper method!